tests/releases/reference_docs_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package references
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/dcl/metadata"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/gvks/supportedgvks"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/servicemapping/servicemappingloader"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/util/repo"
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestReferenceDoc(t *testing.T) {
+	smLoader, err := servicemappingloader.New()
+	if err != nil {
+		t.Fatalf("error creating service mapping loader: %v", err)
+	}
+	serviceMetadataLoader := metadata.New()
+	allGVKs, err := supportedgvks.All(smLoader, serviceMetadataLoader)
+	if err != nil {
+		t.Fatalf("error getting all GVKs: %v", err)
+	}
+
+	var missing []string
+	for _, gvk := range allGVKs {
+		if strings.HasPrefix(gvk.Version, "v1alpha1") {
+			continue
+		}
+		// IAMServiceAccount is a special case, it is not a real resource.
+		if gvk.Kind == "IAMServiceAccount" {
+			continue
+		}
+		if hasReferenceDoc(t, gvk) {
+			continue
+		}
+		missing = append(missing, gvk.String())
+	}
+
+	if len(missing) == 0 {
```

Comment:
This looks like a tests/apichecks test.  Can we follow the pattern there?

Reply:
I was thinking of that at the beginning. But it's for "api" checks, and we want to add more release related checks in this dir.